controller/internal/policy/session.go

```diff
@@ -59,7 +59,10 @@ func (s *SessionEnforcer) Run() error {
 	for i := 0; i < maxIterations; i++ {
 		ids := make([]string, 0, maxDeletePerIteration)
 		err := s.appEnv.GetHandlers().ApiSession.StreamIds(query, func(id string, err error) error {
-			ids = append(ids, id)
+			if lastUpdateAt, hasUnflushedValue := s.appEnv.GetHandlers().ApiSession.HeartbeatCollector.LastUpdatedAt(id); !hasUnflushedValue || lastUpdateAt.Before(oldest) {
```

Comment:
For API Session, deletion use the values from memory over bbolts in case we are between a flush and at the edge of an API Session's life.